Gems/Atom/Asset/ImageProcessingAtom/Code/Source/Compressors/ASTCCompressor.cpp

```diff
@@ -0,0 +1,321 @@
+        case ePixelFormat_R32G32B32A32F:
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    EPixelFormat ASTCCompressor::GetSuggestedUncompressedFormat([[maybe_unused]] EPixelFormat compressedfmt, EPixelFormat uncompressedfmt) const
+    {
+        if (IsUncompressedPixelFormatSupported(uncompressedfmt))
+        {
+            return uncompressedfmt;
+        }
+        
+        auto formatInfo = CPixelFormats::GetInstance().GetPixelFormatInfo(uncompressedfmt);
+        if (formatInfo->eSampleType == ESampleType::eSampleType_Half)
+        {
+            return ePixelFormat_R16G16B16A16F;
+        }
+        else if (formatInfo->eSampleType == ESampleType::eSampleType_Float)
+        {
+            return ePixelFormat_R32G32B32A32F;
+        }
+        return ePixelFormat_R8G8B8A8;
+    }
+
+    ColorSpace ASTCCompressor::GetSupportedColorSpace([[maybe_unused]] EPixelFormat compressFormat) const
+    {
+        return ColorSpace::autoSelect;
+    }
+
+    const char* ASTCCompressor::GetName() const
+    {
+        return "ASTCCompressor";
+    }
+
+    bool ASTCCompressor::DoesSupportDecompress([[maybe_unused]] EPixelFormat fmtDst)
+    {
+        return true;
+    }
+
+    astcenc_profile GetAstcProfile(bool isSrgb, EPixelFormat pixelFormat)
+    {
+        // select profile depends on LDR or HDR, SRGB or Linear
+        //      ASTCENC_PRF_LDR
+        //      ASTCENC_PRF_LDR_SRGB
+        //      ASTCENC_PRF_HDR_RGB_LDR_A
+        //      ASTCENC_PRF_HDR
+        
+        auto formatInfo = CPixelFormats::GetInstance().GetPixelFormatInfo(pixelFormat);
+        bool isHDR = formatInfo->eSampleType == ESampleType::eSampleType_Half || formatInfo->eSampleType == ESampleType::eSampleType_Float;
+        astcenc_profile profile;
+        if (isHDR)
+        {
+            // HDR is not support in core vulkan 1.1 for android.
+            // https://arm-software.github.io/vulkan-sdk/_a_s_t_c.html
+            profile = isSrgb?ASTCENC_PRF_HDR_RGB_LDR_A:ASTCENC_PRF_HDR;
+        }
+        else
+        {
+            
+            profile = isSrgb?ASTCENC_PRF_LDR_SRGB:ASTCENC_PRF_LDR;
+        }
+        return profile;
+    }
+
+    astcenc_type GetAstcDataType(EPixelFormat pixelFormat)
+    {        
+        auto formatInfo = CPixelFormats::GetInstance().GetPixelFormatInfo(pixelFormat);
+        astcenc_type dataType = ASTCENC_TYPE_U8;
+        if (formatInfo->eSampleType == ESampleType::eSampleType_Half)
+        {
+            dataType = ASTCENC_TYPE_F16;
+        }
+        else if (formatInfo->eSampleType == ESampleType::eSampleType_Float)
+        {
+            dataType = ASTCENC_TYPE_F32;
+        }
+        else if (formatInfo->eSampleType == ESampleType::eSampleType_Uint8)
+        {
+            dataType = ASTCENC_TYPE_U8;
+        }
+        else
+        {
+            AZ_Assert(false, "Unsupport uncompressed format %s", formatInfo->szName);
+        }
+
+        return dataType;
```

Comment:
Would this be more efficient and easier to read as a switch statement with immediate returns instead of an if else block?